src/elements/core/base-elements/open-close-escapable-element.ts

```diff
@@ -0,0 +1,57 @@
+import { isServer } from 'lit';
+
+import { SbbOpenCloseBaseElement } from './open-close-base-element.js';
+
+/**
+ * The registry is mapped as an array of WeakRef: it has been noticed that changing page
+ * nor the `disconnectedCallback` nor other `unload` hooks are triggered,
+ * so if the change is done with an opened dialog in the view, its instance is not removed.
+ * This way the reference should be removed from the array when the garbage collector acts.
+ */
+const overlayStack = new Array<WeakRef<SbbOpenCloseEscapableElement>>();
+const registry = new FinalizationRegistry((element: WeakRef<SbbOpenCloseEscapableElement>) => {
+  overlayStack.splice(
+    overlayStack.findIndex((e) => e === element),
+    1,
+  );
+});
+
+if (!isServer) {
+  window.addEventListener('keydown', (event: KeyboardEvent) => {
+    if (event.key === 'Escape' && !event.defaultPrevented) {
+      event.preventDefault();
+      overlayStack.at(-1)?.deref()?.close();
+    }
+  });
+}
+
+/**
+ * Class for overlay components which extends SbbOpenCloseBaseElement
+ * and adds the functionality to close the overlay pressing the `Escape` button.
+ */
+export abstract class SbbOpenCloseEscapableElement extends SbbOpenCloseBaseElement {
```

Comment:
I'm generally insecure whether I consider it a good idea to have a base class for this logic. In our overlay components we use a lot of stuff, like InertController, FocusTrap, ... All of this stuff is implemented as controller or similar which manually gets called at the right moment (like here the call to super.open(). Wouldn't a controller fit better into our existing solutions?

Reply:
The first implementation was with a controller, I could easily revert it